sim/battle.ts

```diff
@@ -2557,6 +2583,8 @@ export class Battle {
 			}
 		}
 
+		if (rotateFirst) return false;
```

Comment:
What should happen if both Pokémon are knocked out, but one side can switch while the other needs to rotate?

Reply:
I'm 90% sure that the auto-rotation should happen first, which is the purpose of this check.

Reply:
Actually I had the wrong use case. What happens if you rotate in and use Pursuit on the Pokémon that the other side is switching? If that uses this code path then the Pursuit executes before the rotation, since it's the `BeforeSwitchOut` event that triggers it.

Reply:
i should have just looked at the old research threads lols https://www.smogon.com/forums/threads/b-w-research-thread.79340/post-3675294 pursuit should not interact if the user has to rotate in first. the code _should_ already be working fine because events are not triggered for sub-active Pokemon, but i'll add a test case for it